In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
def reduce_mem_usage(df):
    """ iterate through all the columns of a dataframe and modify the data type
        to reduce memory usage.        
    """
    start_mem = df.memory_usage().sum() / 1024**2
    print('Memory usage of dataframe is {:.2f} MB'.format(start_mem))

    for col in df.columns:
        col_type = df[col].dtype

        if col_type != object:
            c_min = df[col].min()
            c_max = df[col].max()
            if str(col_type)[:3] == 'int':
                if c_min > np.iinfo(np.int8).min and c_max < np.iinfo(np.int8).max:
                    df[col] = df[col].astype(np.int8)
                elif c_min > np.iinfo(np.int16).min and c_max < np.iinfo(np.int16).max:
                    df[col] = df[col].astype(np.int16)
                elif c_min > np.iinfo(np.int32).min and c_max < np.iinfo(np.int32).max:
                    df[col] = df[col].astype(np.int32)
                elif c_min > np.iinfo(np.int64).min and c_max < np.iinfo(np.int64).max:
                    df[col] = df[col].astype(np.int64)  
            else:
                if c_min > np.finfo(np.float16).min and c_max < np.finfo(np.float16).max:
                    df[col] = df[col].astype(np.float16)
                elif c_min > np.finfo(np.float32).min and c_max < np.finfo(np.float32).max:
                    df[col] = df[col].astype(np.float32)
                else:
                    df[col] = df[col].astype(np.float64)

    end_mem = df.memory_usage().sum() / 1024**2
    print('Memory usage after optimization is: {:.2f} MB'.format(end_mem))
    print('Decreased by {:.1f}%'.format(100 * (start_mem - end_mem) / start_mem))

    return df

In [3]:
pubg_train = reduce_mem_usage(pd.read_csv('train_V2.csv'))

Memory usage of dataframe is 983.90 MB
Memory usage after optimization is: 288.39 MB
Decreased by 70.7%


In [4]:
pubg_train.head()

Id         groupId         matchId  assists  boosts  \
0  7f96b2f878858a  4d4b580de459be  a10357fd1a4a91        0       0   
1  eef90569b9d03c  684d5656442f9e  aeb375fc57110c        0       0   
2  1eaf90ac73de72  6a4a42c3245a74  110163d8bb94ae        1       0   
3  4616d365dd2853  a930a9c79cd721  f1f1f4ef412d7e        0       0   
4  315c96c26c9aac  de04010b3458dd  6dc8ff871e21e6        0       0   

   damageDealt  DBNOs  headshotKills  heals  killPlace  ...  revives  \
0      0.00000      0              0      0         60  ...        0   
1     91.50000      0              0      0         57  ...        0   
2     68.00000      0              0      0         47  ...        0   
3     32.90625      0              0      0         75  ...        0   
4    100.00000      0              0      0         45  ...        0   

   rideDistance  roadKills  swimDistance  teamKills vehicleDestroys  \
0      0.000000          0      0.000000          0               0   
1      0.004501          0     11.039062          0               0   
2      0.000000          0      0.000000          0               0   
3      0.000000          0      0.000000          0               0   
4      0.000000          0      0.000000          0               0   

   walkDistance  weaponsAcquired  winPoints  winPlacePerc  
0        244.75                1       1466      0.444336  
1       1434.00                5          0      0.640137  
2        161.75                2          0      0.775391  
3        202.75                3          0      0.166748  
4         49.75                2          0      0.187500  

[5 rows x 29 columns]

In [5]:
pubg_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4446966 entries, 0 to 4446965
Data columns (total 29 columns):
Id                 object
groupId            object
matchId            object
assists            int8
boosts             int8
damageDealt        float16
DBNOs              int8
headshotKills      int8
heals              int8
killPlace          int8
killPoints         int16
kills              int8
killStreaks        int8
longestKill        float16
matchDuration      int16
matchType          object
maxPlace           int8
numGroups          int8
rankPoints         int16
revives            int8
rideDistance       float16
roadKills          int8
swimDistance       float16
teamKills          int8
vehicleDestroys    int8
walkDistance       float16
weaponsAcquired    int16
winPoints          int16
winPlacePerc       float16
dtypes: float16(6), int16(5), int8(14), object(4)
memory usage: 288.4+ MB


In [6]:
pubg_train['matchType'].unique()

array(['squad-fpp', 'duo', 'solo-fpp', 'squad', 'duo-fpp', 'solo',
       'normal-squad-fpp', 'crashfpp', 'flaretpp', 'normal-solo-fpp',
       'flarefpp', 'normal-duo-fpp', 'normal-duo', 'normal-squad',
       'crashtpp', 'normal-solo'], dtype=object)

In [7]:
solos_df = pubg_train[(pubg_train['matchType'] == 'solo-fpp') | (pubg_train['matchType'] == 'solo') 
                      | (pubg_train['matchType'] == 'normal-solo-fpp')].reset_index().drop('index', axis=1)
solos_df.drop(['groupId', 'matchId', 'DBNOs', 'revives', 'swimDistance', 'walkDistance', 'rideDistance', 'matchType'], axis=1, inplace=True);
solos_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 720387 entries, 0 to 720386
Data columns (total 21 columns):
Id                 720387 non-null object
assists            720387 non-null int8
boosts             720387 non-null int8
damageDealt        720387 non-null float16
headshotKills      720387 non-null int8
heals              720387 non-null int8
killPlace          720387 non-null int8
killPoints         720387 non-null int16
kills              720387 non-null int8
killStreaks        720387 non-null int8
longestKill        720387 non-null float16
matchDuration      720387 non-null int16
maxPlace           720387 non-null int8
numGroups          720387 non-null int8
rankPoints         720387 non-null int16
roadKills          720387 non-null int8
teamKills          720387 non-null int8
vehicleDestroys    720387 non-null int8
weaponsAcquired    720387 non-null int16
winPoints          720387 non-null int16
winPlacePerc       720386 non-null float16
dtypes: float16(3), int16(5), int

In [8]:
solos_df.describe()

assists         boosts  damageDealt  headshotKills  \
count  720387.000000  720387.000000   720387.000  720387.000000   
mean        0.061474       1.059973          NaN       0.248525   
std         0.291209       1.799530          NaN       0.657690   
min         0.000000       0.000000        0.000       0.000000   
25%         0.000000       0.000000        0.000       0.000000   
50%         0.000000       0.000000       74.250       0.000000   
75%         0.000000       2.000000      169.125       0.000000   
max        22.000000      28.000000     6616.000      42.000000   

               heals      killPlace     killPoints          kills  \
count  720387.000000  720387.000000  720387.000000  720387.000000   
mean        1.013490      47.474106     447.321506       0.933884   
std         2.405218      27.723485     595.020288       1.589397   
min         0.000000       1.000000       0.000000       0.000000   
25%         0.000000      23.000000       0.000000       0.000000   
50%         0.000000      47.000000       0.000000       0.000000   
75%         1.000000      71.000000    1090.000000       1.000000   
max        63.000000     100.000000    1970.000000      66.000000   

         killStreaks   longestKill  matchDuration       maxPlace  \
count  720387.000000  720387.00000  720387.000000  720387.000000   
mean        0.474645           NaN    1600.831105      93.942782   
std         0.563667           NaN     256.823622      10.007331   
min         0.000000       0.00000       9.000000       1.000000   
25%         0.000000       0.00000    1384.000000      94.000000   
50%         0.000000       0.00000    1456.000000      96.000000   
75%         1.000000      20.84375    1874.000000      97.000000   
max        18.000000    1002.00000    2237.000000     100.000000   

           numGroups     rankPoints      roadKills      teamKills  \
count  720387.000000  720387.000000  720387.000000  720387.000000   
mean       91.150458     966.801465       0.005115       0.011459   
std        11.317641     744.266943       0.087980       0.106458   
min         1.000000      -1.000000       0.000000       0.000000   
25%        91.000000      -1.000000       0.000000       0.000000   
50%        93.000000    1496.000000       0.000000       0.000000   
75%        95.000000    1517.000000       0.000000       0.000000   
max       100.000000    2857.000000      18.000000       2.000000   

       vehicleDestroys  weaponsAcquired      winPoints   winPlacePerc  
count    720387.000000    720387.000000  720387.000000  720386.000000  
mean          0.004511         3.587341     557.243486            NaN  
std           0.069617         2.594870     728.786109       0.000000  
min           0.000000         0.000000       0.000000       0.000000  
25%           0.000000         2.000000       0.000000       0.229248  
50%           0.000000         3.000000       0.000000       0.483887  
75%           0.000000         5.000000    1492.000000       0.747559  
max           5.000000       153.000000    1922.000000       1.000000

In [9]:
np.nan in solos_df

False

No NaN values in the dataframe, that's good. Should make life easier.

In [10]:
solos_df[solos_df['kills'] >= 50]

Id  assists  boosts  damageDealt  headshotKills  heals  \
54296   810f2379261545       20       0       6616.0             13      5   
124549  2ade4369bccd12       12       0       5556.0              7      4   
128897  7f3edd982813e6        6       0       5408.0             16     14   
202420  80ac0bbf58bfaf        5       0       6376.0             21      4   
236013  c47bd86daa3de6        2       2       4496.0             42      2   
237412  be4ff9afaa5bb1       22       0       5376.0              8      4   
242116  1f6ceaa5140fb6       15       0       5468.0             13      5   
421731  436d1530e9eb00        5       0       4348.0             31      0   
474905  97047ef60af1e7       11       0       5520.0              8      4   

        killPlace  killPoints  kills  killStreaks  ...  matchDuration  \
54296           1           0     65            7  ...           1798   
124549          1           0     55            6  ...           1798   
128897          1        1000     56            5  ...           1797   
202420          1           0     66            8  ...           1390   
236013          1        1000     50            7  ...           1136   
237412          3           0     53            3  ...           1798   
242116          2           0     56            4  ...           1798   
421731          1           0     53            8  ...           1116   
474905          1           0     53            3  ...           1799   

        maxPlace  numGroups  rankPoints  roadKills  teamKills  \
54296         11         11        1500          0          0   
124549        19         18        1500          0          0   
128897        37         22          -1          0          0   
202420        18         12        1500          0          0   
236013        30         17          -1          0          0   
237412        11         11        1500          0          0   
242116        11         11        1500          0          0   
421731        37         26        1500          0          0   
474905        16         14        1500          0          0   

        vehicleDestroys  weaponsAcquired  winPoints  winPlacePerc  
54296                 0               60          0      1.000000  
124549                0               66          0      1.000000  
128897                0               50       1500      1.000000  
202420                0               23          0      1.000000  
236013                0               15       1500      1.000000  
237412                0               83          0      0.799805  
242116                0               60          0      0.899902  
421731                0               36          0      1.000000  
474905                0               57          0      1.000000  

[9 rows x 21 columns]

Having played the game, I'm fairly certain these are cheaters that used hacking techniques to have instant kills across the map. Especially #236013, I don't like that 84% of his kills were headshot kills, and his longest kill was over 300 meters. That screams cheating.

In [11]:
solos_df = solos_df.drop(solos_df[solos_df['kills'] >= 50].index)
solos_df.describe()

assists         boosts  damageDealt  headshotKills  \
count  720378.000000  720378.000000   720378.000  720378.000000   
mean        0.061339       1.059984          NaN       0.248307   
std         0.287729       1.799537          NaN       0.653610   
min         0.000000       0.000000        0.000       0.000000   
25%         0.000000       0.000000        0.000       0.000000   
50%         0.000000       0.000000       74.250       0.000000   
75%         0.000000       2.000000      169.125       0.000000   
max        21.000000      28.000000     5480.000      39.000000   

               heals      killPlace     killPoints          kills  \
count  720378.000000  720378.000000  720378.000000  720378.000000   
mean        1.013444      47.474683     447.324319       0.933192   
std         2.405164      27.723178     595.021659       1.577191   
min         0.000000       1.000000       0.000000       0.000000   
25%         0.000000      23.000000       0.000000       0.000000   
50%         0.000000      47.000000       0.000000       0.000000   
75%         1.000000      71.000000    1090.000000       1.000000   
max        63.000000     100.000000    1970.000000      48.000000   

         killStreaks   longestKill  matchDuration       maxPlace  \
count  720378.000000  720378.00000  720378.000000  720378.000000   
mean        0.474580           NaN    1600.831073      93.943692   
std         0.563333           NaN     256.823257      10.004017   
min         0.000000       0.00000       9.000000       1.000000   
25%         0.000000       0.00000    1384.000000      94.000000   
50%         0.000000       0.00000    1456.000000      96.000000   
75%         1.000000      20.84375    1874.000000      97.000000   
max        18.000000    1002.00000    2237.000000     100.000000   

           numGroups     rankPoints      roadKills      teamKills  \
count  720378.000000  720378.000000  720378.000000  720378.000000   
mean       91.151400     966.798971       0.005115       0.011459   
std        11.314561     744.267990       0.087981       0.106459   
min         1.000000      -1.000000       0.000000       0.000000   
25%        91.000000      -1.000000       0.000000       0.000000   
50%        93.000000    1496.000000       0.000000       0.000000   
75%        95.000000    1517.000000       0.000000       0.000000   
max       100.000000    2857.000000      18.000000       2.000000   

       vehicleDestroys  weaponsAcquired      winPoints   winPlacePerc  
count    720378.000000    720378.000000  720378.000000  720377.000000  
mean          0.004512         3.586761     557.246283            NaN  
std           0.069617         2.588692     728.786899       0.000000  
min           0.000000         0.000000       0.000000       0.000000  
25%           0.000000         2.000000       0.000000       0.229248  
50%           0.000000         3.000000       0.000000       0.483887  
75%           0.000000         5.000000    1492.000000       0.747559  
max           5.000000       153.000000    1922.000000       1.000000

Continuing the search for cheaters, anyone with more than an 80% headshot kill to kill ratio and more than 20 kills could be classified as a cheater and should not be considered within the dataset.

In [12]:
solos_df[((solos_df['headshotKills'] / solos_df['kills']) >= 0.8) & (solos_df['kills'] >= 20)]

Id  assists  boosts  damageDealt  headshotKills  heals  \
158594  8e9eb1ce0e0135        0       7       2068.0             19      2   
330512  3e28994fb2950b        1       0       3696.0             39      2   
544484  a7ad68271a7137        5       0       2560.0             23      0   

        killPlace  killPoints  kills  killStreaks  ...  matchDuration  \
158594          1           0     21            2  ...           1375   
330512          1           0     41            8  ...            886   
544484          3        1000     25            9  ...            897   

        maxPlace  numGroups  rankPoints  roadKills  teamKills  \
158594        97         94        1507          0          0   
330512        17         13        1500          0          0   
544484        21         13          -1          0          0   

        vehicleDestroys  weaponsAcquired  winPoints  winPlacePerc  
158594                0                4          0      1.000000  
330512                0               21          0      1.000000  
544484                0               14       1500      0.899902  

[3 rows x 21 columns]

In [13]:
solos_df = solos_df.drop(solos_df[((solos_df['headshotKills'] / solos_df['kills']) >= 0.8) & (solos_df['kills'] >= 20)].index)

Next we take a look at match duration. Looking for matches that lasted shorter than about 15 minutes could mean that the game started with a well below average number of people.

In [21]:
three_down = np.mean(solos_df['matchDuration']) - 3 * np.std(solos_df['matchDuration'])

print(three_down)

834.5493596376818


So three_down will look for matches in this database that lasted less than 830 seconds, which is about 14 minutes. Games in PUBG typically last around 1600 seconds or around 26 minutes. These are games that could have started with a small amount of people. Only 30 or so people, so getting rid of these will drop all of those games that started with a well below average number of people.

In [15]:
solos_df[(solos_df['matchDuration'] <= three_down)]

Id  assists  boosts  damageDealt  headshotKills  heals  \
211     32220283493bc2        2       0    655.00000              2      0   
1055    6ab29688a68902        0       2     85.68750              0      7   
6004    8358fd4679e2db        4       0    339.00000              1      0   
8206    26809358eab674        1       0    352.25000              0      2   
8415    a16a20a3305521        0       1    522.00000              1      5   
10647   822d53b0965228        4       0   1009.00000              1      0   
16032   a2b88f886138cf        0       0    200.00000              1      1   
16068   1ac82deebad3df        0       0      0.00000              0      0   
16653   d897450a550371        1       1    689.00000              7      3   
20197   ef3c16baa36e0f        0       0     29.40625              0      0   
20239   84184e3526f140        4       2   1791.00000              7      5   
21143   cf4e432326620c        2       0    235.62500              0      0   
21719   45eb268a9fab32        0       1    233.75000              0      0   
24242   f75b8f47b8675f        1       3    310.00000              0      3   
26598   82bfcd83925708        0       0      0.00000              0      0   
26636   2b4f20c20d5736        0       4     58.56250              0      0   
26715   4b7ed1909a8c40        4       0    447.00000              0      0   
27001   84e814e9dc4089        0       3      0.00000              0      2   
29150   053c758519518f        0       0    364.50000              0      0   
30302   523358f9faa3cf        2       1    320.75000              0      1   
31746   46054706d130e3        0       0    262.00000              1      1   
35572   f10a2c460e42cc       10       0   1495.00000              6      0   
38098   44d1bb726b2af4        0       0    100.00000              0      0   
38958   097a38e9a8fd24        3       1   1240.00000              9      0   
39197   9ef33c2fc01625        0       0      0.00000              0      0   
40066   fafe222002361a        1       2    377.50000              2      1   
40350   887443503ee644        0       0    100.00000              0      0   
40595   eef45af6cbdde9        0       0      0.00000              0      0   
41837   82416e142b4e41        0       2    311.50000              0      1   
42424   d7ac32a444ceec        4       0    349.00000              0      0   
...                ...      ...     ...          ...            ...    ...   
683098  1351dd3d39ea67        1       0    276.75000              1      5   
684938  8347567c468db6        2       0    821.50000              3      0   
686178  fef4f27b94c759        0       0     38.25000              0      0   
686560  315faf35951c24        0       0      0.00000              0      0   
686778  94be929addd23b        0       2    250.37500              2      3   
687152  a335977eda225d        0       0      0.00000              0      0   
688989  41e079812e3527        2       1   1480.00000              6      3   
689727  cfd1841d7799a6        0       0    250.25000              1      0   
689759  dc6cbe683efacc        2       0    943.50000              4      0   
690992  3adbd49c0e03a6        0       0    376.50000              2      0   
693896  b3378457853871        3       0    741.00000              3      0   
694631  e0f70bb898eacd        0       0    143.37500              0      0   
694704  9863678ebd5665        0       5    425.50000              0      4   
695410  91bd141d38abe5        0       0    287.50000              1      1   
695609  8e76ffe5ecb88b        2       0    245.87500              0      0   
696174  5fd6166d45bcbd        0       0    240.62500              0      1   
696252  4e20357e59bd17        0       0      0.00000              0      0   
697206  d0db61d38ddf98        2       0    488.00000              0      9   
698383  6c08e29652e3c8        0       0     77.00000              0      0   
699573  39b1f436178dc0        0       0   

In [16]:
solos_df = solos_df.drop(solos_df[(solos_df['matchDuration'] <= three_down)].index)
solos_df.head()

Id  assists  boosts  damageDealt  headshotKills  heals  \
0  315c96c26c9aac        0       0   100.000000              0      0   
1  311b84c6ff4390        0       0     8.539062              0      0   
2  b7807186e3f679        0       1   324.250000              1      5   
3  92022479b92ce7        0       3   254.250000              0     12   
4  47143f942503e0        0       0   136.875000              0      0   

   killPlace  killPoints  kills  killStreaks  ...  matchDuration  maxPlace  \
0         45           0      1            1  ...           1424        97   
1         48        1000      0            0  ...           1967        96   
2          5         986      4            1  ...           1886        97   
3         13           0      2            1  ...           1371        96   
4         37           0      1            1  ...           1425        96   

   numGroups  rankPoints  roadKills  teamKills  vehicleDestroys  \
0         95        1560          0          0                0   
1         92          -1          0          0                0   
2         94          -1          0          0                0   
3         95        1536          0          0                0   
4         94        1500          0          0                0   

   weaponsAcquired  winPoints  winPlacePerc  
0                2          0      0.187500  
1                6       1500      0.736816  
2                6       1462      0.875000  
3                3          0      0.821289  
4                1          0      0.347412  

[5 rows x 21 columns]

In [17]:
solos_df.describe()

assists         boosts  damageDealt  headshotKills  \
count  719937.000000  719937.000000  719937.0000  719937.000000   
mean        0.060916       1.060325          NaN       0.247738   
std         0.285408       1.799856          NaN       0.648956   
min         0.000000       0.000000       0.0000       0.000000   
25%         0.000000       0.000000       0.0000       0.000000   
50%         0.000000       0.000000      74.1875       0.000000   
75%         0.000000       2.000000     169.1250       0.000000   
max        21.000000      28.000000    5480.0000      26.000000   

               heals      killPlace     killPoints          kills  \
count  719937.000000  719937.000000  719937.000000  719937.000000   
mean        1.013297      47.496199     447.310804       0.931598   
std         2.405124      27.716388     595.075446       1.571368   
min         0.000000       1.000000       0.000000       0.000000   
25%         0.000000      23.000000       0.000000       0.000000   
50%         0.000000      47.000000       0.000000       0.000000   
75%         1.000000      71.000000    1090.000000       1.000000   
max        63.000000     100.000000    1970.000000      48.000000   

         killStreaks  longestKill  matchDuration       maxPlace  \
count  719937.000000  719937.0000  719937.000000  719937.000000   
mean        0.474196          NaN    1601.451408      93.987600   
std         0.562466          NaN     255.635701       9.840698   
min         0.000000       0.0000     850.000000       5.000000   
25%         0.000000       0.0000    1384.000000      94.000000   
50%         0.000000       0.0000    1456.000000      96.000000   
75%         1.000000      20.8125    1875.000000      97.000000   
max        18.000000    1002.0000    2237.000000     100.000000   

           numGroups     rankPoints      roadKills      teamKills  \
count  719937.000000  719937.000000  719937.000000  719937.000000   
mean       91.198770     966.903819       0.005117       0.011455   
std        11.153571     744.252849       0.088000       0.106440   
min         1.000000      -1.000000       0.000000       0.000000   
25%        91.000000      -1.000000       0.000000       0.000000   
50%        93.000000    1496.000000       0.000000       0.000000   
75%        95.000000    1517.000000       0.000000       0.000000   
max       100.000000    2857.000000      18.000000       2.000000   

       vehicleDestroys  weaponsAcquired      winPoints   winPlacePerc  
count    719937.000000    719937.000000  719937.000000  719937.000000  
mean          0.004514         3.583739     557.156339            NaN  
std           0.069638         2.580297     728.765535       0.000000  
min           0.000000         0.000000       0.000000       0.000000  
25%           0.000000         2.000000       0.000000       0.229248  
50%           0.000000         3.000000       0.000000       0.483887  
75%           0.000000         5.000000    1492.000000       0.747559  
max           5.000000       153.000000    1922.000000       1.000000

In the teamKills column, there is a max of 2. Someone can't kill themselves twice in a solos game. Let's see that data point

In [18]:
solos_df[solos_df['teamKills'] == max(solos_df['teamKills'])]

Id  assists  boosts  damageDealt  headshotKills  heals  \
58140   6e34b7909e4691        1       2        881.0              6     17   
446343  0c575f045d2fb1        3       0        566.0              1      2   

        killPlace  killPoints  kills  killStreaks  ...  matchDuration  \
58140           6           0     12            2  ...           1097   
446343         23           0      2            1  ...           1093   

        maxPlace  numGroups  rankPoints  roadKills  teamKills  \
58140         11          8        1500          0          2   
446343        23         17        1500          0          2   

        vehicleDestroys  weaponsAcquired  winPoints  winPlacePerc  
58140                 0               38          0      0.399902  
446343                0               26          0      0.000000  

[2 rows x 21 columns]

Two of them. These data points have clearly bad data in them, so they must be removed from the data set

In [19]:
solos_df = solos_df.drop(solos_df[solos_df['teamKills'] == 2].index)
solos_df.describe()

assists         boosts  damageDealt  headshotKills  \
count  719935.000000  719935.000000  719935.0000  719935.000000   
mean        0.060911       1.060325          NaN       0.247729   
std         0.285385       1.799858          NaN       0.648921   
min         0.000000       0.000000       0.0000       0.000000   
25%         0.000000       0.000000       0.0000       0.000000   
50%         0.000000       0.000000      74.1875       0.000000   
75%         0.000000       2.000000     169.1250       0.000000   
max        21.000000      28.000000    5480.0000      26.000000   

               heals      killPlace     killPoints          kills  \
count  719935.000000  719935.000000  719935.000000  719935.000000   
mean        1.013273      47.496291     447.312046       0.931581   
std         2.405053      27.716368     595.075806       1.571316   
min         0.000000       1.000000       0.000000       0.000000   
25%         0.000000      23.000000       0.000000       0.000000   
50%         0.000000      47.000000       0.000000       0.000000   
75%         1.000000      71.000000    1090.000000       1.000000   
max        63.000000     100.000000    1970.000000      48.000000   

         killStreaks  longestKill  matchDuration       maxPlace  \
count  719935.000000  719935.0000  719935.000000  719935.000000   
mean        0.474193          NaN    1601.452814      93.987814   
std         0.562464          NaN     255.634662       9.839870   
min         0.000000       0.0000     850.000000       5.000000   
25%         0.000000       0.0000    1384.000000      94.000000   
50%         0.000000       0.0000    1456.000000      96.000000   
75%         1.000000      20.8125    1875.000000      97.000000   
max        18.000000    1002.0000    2237.000000     100.000000   

           numGroups     rankPoints      roadKills      teamKills  \
count  719935.000000  719935.000000  719935.000000  719935.000000   
mean       91.198989     966.902338       0.005117       0.011450   
std        11.152813     744.253353       0.088000       0.106389   
min         1.000000      -1.000000       0.000000       0.000000   
25%        91.000000      -1.000000       0.000000       0.000000   
50%        93.000000    1496.000000       0.000000       0.000000   
75%        95.000000    1517.000000       0.000000       0.000000   
max       100.000000    2857.000000      18.000000       1.000000   

       vehicleDestroys  weaponsAcquired      winPoints   winPlacePerc  
count    719935.000000    719935.000000  719935.000000  719935.000000  
mean          0.004514         3.583660     557.157886            NaN  
std           0.069638         2.579846     728.765955       0.000000  
min           0.000000         0.000000       0.000000       0.000000  
25%           0.000000         2.000000       0.000000       0.229248  
50%           0.000000         3.000000       0.000000       0.483887  
75%           0.000000         5.000000    1492.000000       0.747559  
max           5.000000       153.000000    1922.000000       1.000000

In [20]:
solos_df = solos_df.reset_index().drop('index', axis=1)